sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -1013,20 +1047,16 @@ private async Task RunAsync(CancellationToken cancellationToken)
             // avoid overlooking this kind of situation, we may want to claim an ownership when we have exactly the minimum amount of ownership,
             // but we are making sure there are no better candidates among the other event processors.
 
-            if (ownedPartitionsCount < minimumOwnedPartitionsCount
-                || ownedPartitionsCount == minimumOwnedPartitionsCount
-                && !partitionDistribution.Values.Any(partitions => partitions < minimumOwnedPartitionsCount))
+            if (ownedPartitionsCount < minimumOwnedPartitionsCount ||
+                (ownedPartitionsCount == minimumOwnedPartitionsCount &&
+                    !ActiveOwnershipWithDistribution.Values.Any(partitions => partitions.Count < minimumOwnedPartitionsCount)))
```

Comment:
I don't have a strong opinion, but I do think that splitting the compound expression makes it a bit tougher to parse at a glance.  I'd suggest something like:

```csharp
if ((ownedPartitionsCount < minimumOwnedPartitionsCount)
    || ((ownedPartitionsCount == minimumOwnedPartitionsCount) && (!ActiveOwnershipWithDistribution.Values.Any(partitions => partitions.Count < minimumOwnedPartitionsCount)))
{
...
}
```

But, I tend to work in a resolution where longer lines aren't an impediment and explicitly parenthesize things that others do not, so....

If the line length is a concern, this is a case where it may be helpful to store the intermediate expressions outside of the `if`.